runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -4107,6 +4123,32 @@ J9::Z::CodeGenerator::inlineDirectCall(
                break;
             }
          }
+      if (cg->getSupportsVectorRegisters())
+         {
+         switch (methodSymbol->getRecognizedMethod())
+            {
+            case TR::java_lang_Math_fma_D:
+            case TR::java_lang_StrictMath_fma_D:
+               resultReg = inlineMathFma(node, cg);
+               return true;
+
+            case TR::java_lang_Math_fma_F:
+            case TR::java_lang_StrictMath_fma_F:
+               if (comp->target().cpu.getSupportsVectorFacilityEnhancement1())
+                  {
+                  resultReg = inlineMathFma(node, cg);
+                  return true;
+                  }
+               else 
```

Comment:
Is this what we want here since it would stop it being recognized later? Would it make more sense to follow the convention an inner if with another switch for these two methods where the resultReg would be set and return true would happen so that if it does not match or the capability is not enabled evaluation will continue like for most other matchers?

Reply:
My intention for having an `else` to return false here is to save time from checking any other recognized methods. Because if the method falls into the above cases but does not pass the cpu capability check, it shouldn't be matched to any other recognized method.

Reply:
@fjeremic Are you happy with the `return false` coding style here? Replacing the else statement by a simple `break` might also be a good practice as there might be related recognized method checks afterwards (especially in this extensible classes hierarchy).

Reply:
> Because if the method falls into the above cases but does not pass the cpu capability check, it shouldn't be matched to any other recognized method.

Let's not assume this. This class is extensible, and there could be other checks further down. I'd remove the `else` block entirely.

Reply:
Fixed